src/avram/uploadable.cr

```diff
@@ -0,0 +1,54 @@
+# Include this module in classes that represent an uploaded file from a form
+module Avram::Uploadable
+  getter name : String
+  getter tempfile : File
+  getter metadata : HTTP::FormData::FileMetadata
```

Comment:
Do you think it might be possible to abstract this metadata somewhat?

I love your approach of being able to use this module outside of `Lucky::UploadedFile`, but this means it could be used for files that arrived some other way, not necessarily via HTTP. These could of course simply construct their own `HTTP::FormData::FileMetadata`, but it would feel a bit awkward imho.

Also, avram is supposed to be a general purpose ORM. Seeing things related to HTTP here is at least a little bit confusing.

Maybe it might suffice to simply add the attributes from `HTTP::FormData::FileMetadata` that are relevant here directly (possibly size and the different timestamps). Personally I would be fine with removing this altogether. At this point we already have the complete file to determine the size reliably and I have never looked at what timestamps the browser sends with files before.

Reply:
Very good point! We only need it for the original `filename`. I think the `metadata` property can move back to `Lucky::UploadedFile` and `Avram::Uploadable` can declare an interface for `filename`:

```cr
module Avram::Uploadable
  abstract filename : String
  # ...
end
```

So that includers can implement it however they like:

```cr
module Lucky::UploadedFile
  getter metadata : HTTP::FormData::FileMetadata
  # ...
  def filename : String
    metadata.filename.to_s
  end
end
```

I think we'll also need to rethink the `name` property as well (as discussed in https://github.com/luckyframework/lucky/issues/1147). I can't seem to find a place where it's actually used. For clarity, at least we could change it to `part_name`.